frontend/src/metabase/components/SingleSelectListField/SingleSelectListField.tsx

```diff
@@ -0,0 +1,147 @@
+  const [selectedValue, setSelectedValue] = useState(value?.[0]);
+  const [addedOptions, setAddedOptions] = useState<Option>(() =>
+    createOptionsFromValuesWithoutOptions(value, options),
+  );
+
+  const augmentedOptions = useMemo(() => {
+    return [...options.filter(option => option[0] != null), ...addedOptions];
+  }, [addedOptions, options]);
+
+  const sortedOptions = useMemo(() => {
+    if (selectedValue) {
+      return augmentedOptions;
+    }
+
+    const [selected, unselected] = _.partition(
+      augmentedOptions,
+      option => selectedValue === option[0],
+    );
+
+    return [...selected, ...unselected];
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [augmentedOptions.length]);
+
+  const [filter, setFilter] = useState("");
+  const debouncedFilter = useDebouncedValue(filter, SEARCH_DEBOUNCE_DURATION);
+
+  const filteredOptions = useMemo(() => {
+    const formattedFilter = debouncedFilter.trim().toLowerCase();
+    if (formattedFilter.length === 0) {
+      return sortedOptions;
+    }
+
+    return augmentedOptions.filter(option => {
+      if (!option || option.length === 0) {
+        return false;
+      }
+
+      // option as: [id, name]
+      if (
+        option.length > 1 &&
+        option[1] &&
+        isValidOptionItem(option[1], formattedFilter)
+      ) {
+        return true;
+      }
+
+      // option as: [id]
+      return isValidOptionItem(option[0], formattedFilter);
+    });
+  }, [augmentedOptions, debouncedFilter, sortedOptions]);
+
+  const shouldShowEmptyState =
+    augmentedOptions.length > 0 && filteredOptions.length === 0;
+
+  const onClickOption = (option: any) => {
+    if (selectedValue !== option) {
+      setSelectedValue(option);
+      onChange?.([option]);
+    }
+  };
+
+  const handleKeyDown = (event: React.KeyboardEvent) => {
+    if (
+      event.key === "Enter" &&
+      !_.find(augmentedOptions, option => option[0] === filter)
+    ) {
+      setAddedOptions([...addedOptions, [filter]]);
+    }
+  };
+
+  return (
+    <>
+      <FilterInput
+        isDashboardFilter={isDashboardFilter}
+        padding={isDashboardFilter ? "md" : "sm"}
+        borderRadius={isDashboardFilter ? "md" : "sm"}
+        colorScheme={isDashboardFilter ? "transparent" : "admin"}
+        placeholder={placeholder}
+        value={filter}
+        onChange={setFilter}
+        onKeyDown={handleKeyDown}
+        hasClearButton
+        autoFocus
+      />
+
+      {shouldShowEmptyState && (
+        <EmptyStateContainer>
+          <EmptyState message={t`Didn't find anything`} icon="search" />
+        </EmptyStateContainer>
+      )}
+
+      <OptionsList isDashboardFilter={isDashboardFilter}>
+        {filteredOptions.map(option => (
+          <OptionContainer key={option[0]}>
+            <OptionItem
+              data-testid={`${option[0]}-filter-value`}
+              selectedColor={
+                checkedColor ?? isDashboardFilter ? "brand" : "filter"
```

Comment:
This `selectedColor` expression is probably wrong, I just blindly copied it the `checkedColor` value for ListField [here](https://github.com/metabase/metabase/blob/d05aa0b24c25598bcc8566f43bb659a6f5175213/frontend/src/metabase/components/ListField/ListField.tsx#L134)